### PR TITLE
Make python-3.11 and python-3.12 co-installable.

### PR DESCRIPTION
**Pull request by smoser**
Fixes #2115

Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [ ] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [ ] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: python-3.11.yaml [hunk 1]
```diff
@@ -93,6 +88,21 @@ test:
...
+    description: "python-3.11 as /usr/bin/python3"
+    dependencies:
+      provides:
+        - python3=3.11.999
+        - python-3=3.11.999
+      runtime:
+        - python-3.11
+    pipeline:
+      - runs: |
+          mkdir ${{targets.subpkgdir}}/usr/bin
```
Comment (xnox) [inline python-3.11.yaml:101]: `-p` please

File: python-3.12.yaml [hunk 2]
```diff
@@ -93,7 +88,22 @@ test:
...
+    description: "python-3.12 as /usr/bin/python3"
+    dependencies:
+      provides:
+        - python3=${{package.full-version}}
+        - python-3=${{package.full-version}}
+      runtime:
+        - python-3.12
+    pipeline:
+      - runs: |
+          mkdir ${{targets.subpkgdir}}/usr/bin
```
Comment (xnox) [inline python-3.12.yaml:101]: `-p` please

File: python-3.12.yaml [hunk 3]
```diff
@@ -106,6 +116,8 @@ subpackages:
           # pyconfig.h is needed at runtime... ugh.
           mkdir -p "${{targets.destdir}}"/usr/include/python3.12
           mv "${{targets.subpkgdir}}"/usr/include/python3.12/pyconfig.h "${{targets.destdir}}"/usr/include/python3.12
+          mkdir -p "${{targets.subpkgdir}}"/usr/lib
+          mv ${{targets.destdir}}/usr/lib/libpython3.so "${{targets.subpkgdir}}"/usr/lib
```
Comment (xnox) [inline python-3.12.yaml:120]: -dev packages must be co-installable too for compiled extensions.

Please move this to the -default sub package.

File: python-3.11.yaml [hunk 4]
```diff
@@ -106,6 +116,8 @@ subpackages:
           # pyconfig.h is needed at runtime... ugh.
           mkdir -p "${{targets.destdir}}"/usr/include/python3.11
           mv "${{targets.subpkgdir}}"/usr/include/python3.11/pyconfig.h "${{targets.destdir}}"/usr/include/python3.11
+          mkdir -p "${{targets.subpkgdir}}"/usr/lib
+          mv ${{targets.destdir}}/usr/lib/libpython3.so "${{targets.subpkgdir}}"/usr/lib
```
Comment (xnox) [inline python-3.11.yaml:120]: -dev packages must be co-installable too.

Please move it to the -default sub package for now.

**Review by xnox — CHANGES_REQUESTED** [review]
Very good! Just small tweaks left before landing.

File: python-3.11.yaml (hunk 4 above)
Comment (smoser) [reply python-3.11.yaml:120]: bah. fixing. thanks.

File: python-3.12.yaml (hunk 2 above)
Comment (smoser) [reply python-3.12.yaml:101]: i don't know. -p not necessary because subpkgdir starts out empty.... hm... but why did this work? i'll add -p

Comment (xnox) [reply python-3.12.yaml:101]: Ci builds pipeline failed as /usr/ is not there and it's like no such directory.

File: python-3.12.yaml (hunk 3 above)
Comment (smoser) [reply python-3.12.yaml]: I think I'm just going to remove it for now.

**Review by xnox — APPROVED** [review]

**Pull request merged**